src/main/scala/com/datastax/sparkstress/ReadTask.scala

```diff
@@ -48,18 +59,169 @@ abstract class ReadTask(config: Config, sc: SparkContext) extends StressTask {
 }
 
 /**
+  * Full Table Scan One Column using DataSets
+  * Performs a full table scan but only retrieves a single column from the underlying
+  * table.
+  */
+class FTSOneColumn_DS(config: Config, ss: SparkSession) extends ReadTask(config, ss) {
+
+  def run(): Unit = {
+    //val colorCounts = ss.read.cassandraFormat(table, keyspace).load().select("color").count
+    val count = ss.read
+      .format("org.apache.spark.sql.cassandra")
+      .options(Map("table" -> table, "keyspace" -> keyspace))
+      .load()
+      .select("color")
+      .count
+    println(s"Loaded $count rows")
+  }
+}
+
+/**
+  * Full Table Scan Two Columns using DataSets
+  * Performs a full table scan but only retrieves a single column from the underlying
+  * table.
+  */
+class FTSTwoColumns_DS(config: Config, ss: SparkSession) extends ReadTask(config, ss) {
+
+  def run(): Unit = {
+    val count = ss.read
+      .format("org.apache.spark.sql.cassandra")
+      .options(Map("table" -> table, keyspace -> "keyspace"))
+      .load()
+      .select("color", "size")
+      .count
+    println(s"Loaded $count rows")
+  }
+}
+
+/**
+  * Full Table Scan Three Columns using DataSets
+  * Performs a full table scan but only retrieves a single column from the underlying
+  * table.
+  */
+class FTSThreeColumns_DS(config: Config, ss: SparkSession) extends ReadTask(config, ss) {
+
+  def run(): Unit = {
+    val count = ss.read
+      .format("org.apache.spark.sql.cassandra")
+      .options(Map("table" -> table, keyspace -> "keyspace"))
+      .load()
+      .select("color", "size", "qty")
+      .count
+    println(s"Loaded $count rows")
+  }
+}
+
+/**
+  * Full Table Scan Four Columns using DataSets
+  * Performs a full table scan but only retrieves a single column from the underlying
+  * table.
+  */
+class FTSFourColumns_DS(config: Config, ss: SparkSession) extends ReadTask(config, ss) {
+
+  def run(): Unit = {
+    val count = ss.read
+      .format("org.apache.spark.sql.cassandra")
+      .options(Map("table" -> table, keyspace -> "keyspace"))
+      .load()
+      .select("color", "size", "qty", "order_number")
```

Comment:
to avoid duplication we could have a common supertype, that encapsulates reading logic, with columns-to-select parameter